packages/go_router/example/lib/stateful_shell_route.dart

```diff
@@ -168,19 +168,17 @@ class ScaffoldWithNavBar extends StatelessWidget {
   /// Navigate to the current location of the branch at the provided index when
   /// tapping an item in the BottomNavigationBar.
   void _onTap(BuildContext context, int index) {
-    if (index != navigationShell.currentIndex) {
-      // When navigating to a new branch, it's recommended to use the goBranch
-      // method, as doing so makes sure the last navigation state of the
-      // Navigator for the branch is restored.
-      navigationShell.goBranch(index);
-    } else {
-      // If tapping the bar item for the branch that is already active, go to
-      // the initial location instead. The method
-      // effectiveInitialBranchLocation can be used to get the (implicitly or
-      // explicitly) configured initial location in a convenient way.
-      GoRouter.of(context)
-          .go(navigationShell.effectiveInitialBranchLocation(index));
-    }
+    // When navigating to a new branch, it's recommended to use the goBranch
+    // method, as doing so makes sure the last navigation state of the
+    // Navigator for the branch is restored.
+    navigationShell.goBranch(
+      index,
+      // A common pattern when using bottom navigation bars is to support
+      // navigating to the initial location when tapping the item that is
+      // already active. This example demonstrates how to support this behavior,
+      // using the resetLocation parameter of goBranch.
+      resetLocation: index == navigationShell.currentIndex,
```

Comment:
Wouldn't this be `resetLocation: true`?

Reply:
In my earlier comment I meant that  the location wouldn't reset when the index is different from currentIndex.

Reply:
In the latest code, `goBranch` is called each time a bar item is tapped. Normally when tapping an item, you'd want to go to the current state of that index. Say that you are on the first branch and tap on the item for the second, then `index` in `_onTap` would be `1` and `navigationShell.currentIndex would` be `0`, i.e. `resetLocation: false`. But if tapping the index that is already active (i.e. `index == navigationShell.currentIndex`), you want to reset the location (`resetLocation: true`) and go to the initial location of the branch.

Let me know if I misunderstood you somehow (a bit tired right now so entirely possible 😄).

Reply:
I see, I misunderstood how this worked. This makes sense to me.